### PR TITLE
Exception handling for 7z entries without timestamps

### LINKED ISSUE #3035: UnsupportedOperationException on SevenZArchiveEntry.getLastModifiedDate()
## Exception
* __App Name:__ Amaze File Manager
* __Package:__ com.amaze.filemanager
* __Version:__ 3.6.7
* __User Action:__ UI Error
* __Request:__ Application crash
* __OS:__ Linux Android 11 - 30
* __Device:__ elsa
* __Model:__ LG-H918
* __Product:__ elsa_tmo_us
<details><summary><b>Crash log </b></summary><p>

```
java.lang.RuntimeException: An error occurred while executing doInBackground()
	at android.os.AsyncTask$4.done(AsyncTask.java:415)
	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
	at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
	at java.util.concurrent.FutureTask.run(FutureTask.java:271)
	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:305)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:923)
Caused by: java.lang.UnsupportedOperationException: The entry doesn't have this timestamp
	at com.amaze.filemanager.filesystem.compressed.sevenz.SevenZArchiveEntry.getLastModifiedDate(SevenZArchiveEntry.java:207)
	at com.amaze.filemanager.filesystem.compressed.extractcontents.helpers.SevenZipExtractor.extractEntry(SevenZipExtractor.kt:119)
	at com.amaze.filemanager.filesystem.compressed.extractcontents.helpers.SevenZipExtractor.extractWithFilter(SevenZipExtractor.kt:75)
	at com.amaze.filemanager.filesystem.compressed.extractcontents.Extractor.extractEverything(Extractor.java:82)
	at com.amaze.filemanager.asynchronous.services.ExtractService$DoWork.doInBackground(ExtractService.java:318)
	at com.amaze.filemanager.asynchronous.services.ExtractService$DoWork.doInBackground(ExtractService.java:227)
	at android.os.AsyncTask$3.call(AsyncTask.java:394)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	... 4 more

```
</details>
<hr>

### PR DESCRIPTION
**Pull request by TranceLove**
It seems [such archives can be created intentionally](https://sourceforge.net/p/sevenzip/discussion/45797/thread/61905a4c/).

For 7z archives having entries with no timestamp,
- they will show as 00:00:00 1970-01-01 when in file manager view
- they will have last modified date = current timestamp on extracting

#### Issue tracker   
Fixes #3035

#### Automatic tests
- [x] Added test cases
  
#### Manual tests
- [x] Done  
  
- Device: Pixel 2 emulator
- OS: Android 11

#### Build tasks success  

Successfully running following tasks on local:
- [x] `./gradlew assembledebug`
- [x] `./gradlew spotlessCheck`

### CODE REVIEW

**Review by VishalNehra — APPROVED** [review]

File: app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/compress/SevenZipHelperCallable.kt [hunk 1]
```diff
@@ -64,7 +63,7 @@ class SevenZipHelperCallable(
                     elements.add(
                         CompressedObjectParcelable(
                             entry.name,
-                            entry.lastModifiedDate.time,
+                            runCatching { entry.lastModifiedDate.time }.getOrElse { 0L },
```
Comment (EmmanuelMess) [inline app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/compress/SevenZipHelperCallable.kt:66]: What does this catch? This should be more explicit of what it does.

**Review by EmmanuelMess — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/amaze/filemanager/filesystem/compressed/extractcontents/helpers/SevenZipExtractor.kt [hunk 2]
```diff
@@ -126,7 +125,10 @@ class SevenZipExtractor(
                     progress += length.toLong()
                 }
                 close()
-                outputFile.setLastModified(entry.lastModifiedDate.time)
+                val lastModifiedDate = runCatching {
+                    entry.lastModifiedDate.time
+                }.getOrElse { System.currentTimeMillis() }
```
[the remark this answers is not in the document]
Comment (EmmanuelMess) [reply app/src/main/java/com/amaze/filemanager/filesystem/compressed/extractcontents/helpers/SevenZipExtractor.kt:130]: Can't we have an "unknown" state?

[the remark this answers is not in the document]
Comment (VishalNehra) [reply app/src/main/java/com/amaze/filemanager/filesystem/compressed/extractcontents/helpers/SevenZipExtractor.kt]: can't seem like we can have

**Review by VishalNehra — APPROVED** [review]

File: app/src/main/java/com/amaze/filemanager/filesystem/compressed/extractcontents/helpers/SevenZipExtractor.kt [hunk 3]
```diff
@@ -126,7 +127,13 @@ class SevenZipExtractor(
                     progress += length.toLong()
                 }
                 close()
-                outputFile.setLastModified(entry.lastModifiedDate.time)
+                val lastModifiedDate = try {
+                    entry.lastModifiedDate.time
+                } catch (e: UnsupportedOperationException) {
+                    Log.w(javaClass.simpleName, "Unable to get modified date for 7zip file")
```
Comment (EmmanuelMess) [inline app/src/main/java/com/amaze/filemanager/filesystem/compressed/extractcontents/helpers/SevenZipExtractor.kt:133]: These should use `TAG` why was this fixed like this?

**Pull request merged**